core/modules/map.rs

```diff
@@ -279,6 +279,12 @@ impl ModuleMap {
       module_url_specified
     };
 
+    // TODO(bartlomieju): I have a hunch that this is wrong - write a test
+    // that tries to "confuse" the type system, by first requesting a module
+    // with type `RequestedModuleType::Other("foo".into)``, and then the loader
+    // actually returns `ModuleType::Other("bar".into())`. See if it leads to
+    // unexpected result in how `ModuleMap` is structured and verify how
+    // querying the module map works (`ModuleMap::get_by_id`, `ModuleMap::get_by_name`).
     let requested_module_type = RequestedModuleType::from(module_type.clone());
```

Comment:
Doesn't the recursive load code check this?

```
    if requested_module_type != module_source.module_type {
      return Err(ModuleError::Other(generic_error(format!(
        "Expected a \"{}\" module but loaded a \"{}\" module.",
        requested_module_type, module_source.module_type,
      ))));
    }
```

Reply:
That code was removed in https://github.com/denoland/deno_core/commit/7fd84216ffdcb331f4ba38ca133b5c2ce31d0c3c.